### ISSUE TITLE
False positive of no-param-reassign on destruction assigment

### ISSUE THREAD

**Issue by tinovyatkin** [eslint/eslint#8279]
**Tell us about your environment**

* **ESLint Version:** 3.18
* **Node Version:** 7.7.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint no-param-reassign: ["error", {"props": true}] */

const { hasOwnProperty: has } = Object.prototype;
module.exports.hasDeepProperty = (obj, propertyPath) => {
  if (!obj) return false;

  const properties = propertyPath.split('.');
  let o = obj;
  return !properties.some((propertyName) => {
    if (!has.call(o, propertyName)) return true;
    ({ [propertyName]: o } = o);
    return false;
  });
};
```

**What did you expect to happen?**
Nothing, no errors

**What actually happened? Please include the actual, raw output from ESLint.**
`no-param-reassign` thinks I'm reassigning `propertyName` at destruction

**Comment by not-an-aardvark** [comment]
I can reproduce this, thanks for the report.

I've edited the issue to include the configuration that causes this issue to appear.
